src/main/java/it/pagopa/pn/deliverypush/middleware/queue/consumer/handler/utils/HandleEventUtils.java

```diff
@@ -0,0 +1,37 @@
+package it.pagopa.pn.deliverypush.middleware.queue.consumer.handler.utils;
+
+import it.pagopa.pn.api.dto.events.StandardEventHeader;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.messaging.MessageHeaders;
+
+import java.time.Instant;
+
+import static it.pagopa.pn.api.dto.events.StandardEventHeader.*;
+
+@Slf4j
+public class HandleEventUtils {
+    private HandleEventUtils() {}
+
+    public static void handleException(MessageHeaders headers, Exception ex) {
+        if(headers != null){
+            StandardEventHeader standardEventHeader = mapStandardEventHeader(headers);
+            log.error("Generic exception for iun={} ex={}", standardEventHeader.getIun(), ex);
+        }else {
+            log.error("Generic exception ex ", ex);
+        }
+    }
+    
+    public static StandardEventHeader mapStandardEventHeader(MessageHeaders headers) {
+        return StandardEventHeader.builder()
+                .eventId((String) headers.get(PN_EVENT_HEADER_EVENT_ID))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;headers&#39; (index: 0 | type: MessageHeaders). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other methods validating the parameter: &#39;handleException&#39;. The same parameter type is validated here for example: pn-delivery-push/src/main/java/it/pagopa/pn/deliverypush/middleware/queue/consumer/handler/utils/HandleEventUtils.java:16. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).